xbmc/utils/StringUtils.h

```diff
@@ -79,13 +79,13 @@ class StringUtils
   \return Formatted string
   */
   template<typename... Args>
-  static constexpr std::string Format(std::string_view format, Args&&... args)
```

Comment:
Why removing the `constexpr`?

Reply:
@neo1973 can `std::string` ever be a `constexpr` ?

Reply:
Yes, since C++20. [Here a very constructed example how that could look](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:21,endLineNumber:12,positionColumn:21,positionLineNumber:12,selectionStartColumn:21,selectionStartLineNumber:12,startColumn:21,startLineNumber:12),source:'%23include+%3Cstring%3E%0A%0Aconstexpr+std::string+first_word_to_str(const+char*+s)+%7B%0A++++std::string+str(s)%3B%0A++++int+pos+%3D+str.find_first_of(!'+!')%3B%0A++++return+%7Bstr.begin(),+str.begin()+%2B+pos%7D%3B%0A%0A%7D%0A%0Aconsteval+int+len_first_word(const+char*+c_str)+%7B%0A++++std::string+s+%3D+first_word_to_str(c_str)%3B%0A++++return+s.size()%3B%0A%7D%0A%0Aint+main()+%7B%0A++++return+len_first_word(%22Hello+World!!%22)%3B%0A%7D'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g142,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-std%3Dc%2B%2B20',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+14.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+14.2+(Compiler+%231)',t:'0')),k:33.33333333333333,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).

Reply:
@neo1973 have we switched to c++20 already?

Reply:
We did, see #24900 :slightly_smiling_face:

Reply:
Okay.